ksm.go

```diff
@@ -0,0 +1,303 @@
+//
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package main
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
+type ksmSetting struct {
+	pagesPerScanFactor int64
+	scanIntervalMs     uint32
+}
+
+type ksmSettingKnob string
+
+const (
+	ksmSlug       ksmSettingKnob = "slug"
+	ksmSlow                      = "slow"
+	ksmStandard                  = "standard"
+	ksmAggressive                = "aggressive"
+)
+
+func (k ksmSettingKnob) String() string {
+	switch k {
+	case ksmSlug:
+		return "slug"
+	case ksmSlow:
+		return "slow"
+	case ksmStandard:
+		return "standard"
+	case ksmAggressive:
+		return "aggressive"
+	}
+
+	return ""
```

Comment:
Will this scenario ever be hit, or could this function default to slug or standard?